ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +54,39 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    # We need to check that Fastapi instrumentation hasn't already unpatched these
+    if isinstance(starlette.routing.Route.handle, ObjectProxy):
+        _u(starlette.routing.Route, "handle")
+
+    if isinstance(starlette.routing.Mount.handle, ObjectProxy):
+        _u(starlette.routing.Mount, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    # Since handle can be called multiple times for one request, we take the path of each instance
+    # Then combine them at the end to get the correct resource name
+    scope = get_argument_value(args, kwargs, 0, "scope")
+    if "__dd_paths__" in scope:
+        scope["__dd_paths__"].append(instance.path)
+
+    else:
+        scope["__dd_paths__"] = [instance.path]
+
+    span = tracer.current_root_span()
+    # Update root span resource
+    if span:
+        # We want to make sure that the root span we grab is the one created from automatic instrumentation
+        # And not a custom root span
+        if span.name == "fastapi.request" or span.name == "starlette.request":
```

Comment:
Thinking more about this check, it is quite brittle (ex. what if the user wants to modify the operation of these spans?). I wonder if we should be storing the fastapi/starlette request span in the asgi scope and then retrieving it here?